### PR TITLE
Product variants enhancements

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because:
- it removes docking from savebar
- hides attribute section if there are no attributes to display

**PR intended to be tested with API branch:** main

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6120 Remove Variant attributes with 0 options](https://app.clickup.com/t/2549495/SALEOR-6120)

**Comment by patrys** [comment]
Task linked: [SALEOR-6119 Save panel should be floating in entity editing screen](https://app.clickup.com/t/2549495/SALEOR-6119)

File: src/products/components/ProductVariantPage/ProductVariantPage.tsx [hunk 1]
```diff
@@ -223,175 +223,190 @@ const ProductVariantPage: React.FC<ProductVariantPageProps> = ({
...
-                    entityId={variant?.id}
-                    title={intl.formatMessage(
-                      messages.selectionAttributesHeader
-                    )}
-                    attributes={data.attributes.filter(
-                      attribute =>
-                        attribute.data.variantAttributeScope ===
-                        VariantAttributeScope.VARIANT_SELECTION
+          }) => {
+            const nonSelectionAttributes = data.attributes.filter(
```
Comment (bmigirl) [inline src/products/components/ProductVariantPage/ProductVariantPage.tsx:227]: `extractVariantSelectionAttributes` and `extractNotVariantSelectionAttributtes` plz

**Review by bmigirl — CHANGES_REQUESTED** [review]

File: src/products/components/ProductVariantPage/ProductVariantPage.tsx [hunk 2]
```diff
@@ -223,175 +223,190 @@ const ProductVariantPage: React.FC<ProductVariantPageProps> = ({
...
+            );
+
+            return (
+              <>
+                <Grid variant="inverted">
+                  <div>
+                    <ProductVariantNavigation
+                      current={variant ? variant.id : undefined}
+                      defaultVariantId={defaultVariantId}
+                      fallbackThumbnail={maybe(
```
Comment (jwm0) [inline src/products/components/ProductVariantPage/ProductVariantPage.tsx:245]: Remove all use of `maybe` fn in this file

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/products/components/ProductVariantPage/ProductVariantPage.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/products/components/ProductVariantPage/ProductVariantPage.tsx:227]: Why? It's simple and non-reusable filter. Proposed function's name is almost as long as function body.

**Review by jwm0 — APPROVED** [review]

File: src/products/components/ProductVariantPage/ProductVariantPage.tsx (hunk 1 above)
Comment (bmigirl) [reply src/products/components/ProductVariantPage/ProductVariantPage.tsx:226]: Simply for the sake of readability. I think as a general rule if we need to extract consts from the places they're being used in the jsx then putting them somehwere else in the jsx makes it less readable and clean than if we placed them outside of return

**Review by Cloud11PL — APPROVED** [review]

File: src/products/components/ProductVariantPage/ProductVariantPage.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/products/components/ProductVariantPage/ProductVariantPage.tsx:226]: Its readability gain is close to none, but have it your way

**Review by karola312 — APPROVED** [review]

**Review by bmigirl — APPROVED** [review]

**Pull request merged**